src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        array = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
+        if (size == array.length) {
+            grow();
+        }
 
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
Don't begin class or method implementation with an empty line. Remove the empty line at the start of the 'add' method with an index.